app/ui/client/lib/UserAction.js

```diff
@@ -0,0 +1,162 @@
+import { Meteor } from 'meteor/meteor';
+import { Tracker } from 'meteor/tracker';
+import { ReactiveDict } from 'meteor/reactive-dict';
+import { Session } from 'meteor/session';
+import _ from 'underscore';
+
+import { settings } from '../../../settings';
+import { Notifications } from '../../../notifications';
+
+const timeout = 15000;
+const renew = timeout / 3;
+const rooms = {};
+
+export const USER_RECORDING = 'user-recording';
+export const USER_TYPING = 'user-typing';
+export const USER_UPLOADING = 'user-uploading';
+export const USER_ACTIVITY = 'user-activity';
+
+
+const activityTimeouts = {};
+const activityRenews = {};
```

Comment:
Use Map for those structures

Reply:
This was not resolved. My suggestion was for you to use for example `const activityRenews = new Map();`

Reply:
okay.. But what is the idea behind it? Why would you prefer a map over an object here?